File: DevEnvironment/Base/src/main/java/base/core/WindowsPathSettings.java
package base.core;

import base.interfaces.IOperatingSystemPathSettings;
import base.interfaces.IPath;
import base.interfaces.IPathBuilder;

/**
 * The WindowsPathSettings class implements path settings of a windows operating system.
 */
public final class WindowsPathSettings implements IOperatingSystemPathSettings {
    /**
     * The WindowsPathSettings constructor.
     */
    public WindowsPathSettings() {
    }

    /**
     * Gets a path separator.
     */
    @Override
    public String getPathSeparator() {
        return WindowsPath.DIRECTORY_SEPARATOR;
    }

    /**
     * Creates a path.
     */
    @Override
    public IPath createPath(String path) {
        return new WindowsPath(path);
    }

    /**
     * Creates a path builder.
     */
    @Override
    public IPathBuilder createPathBuilder() {
        return new PathBuilder();
    }
}

File: SystemCore/Base/core/WindowsPathSettings.cpp
#include "WindowsPathSettings.h"

#include "PathBuilder.h"
#include "WindowsPath.h"

using namespace base;
using namespace base::environment::windows;

/**
 * The WindowsPathSettings constructor.
 */
WindowsPathSettings::WindowsPathSettings() {
}

/**
 * The WindowsPathSettings destructor.
 */
WindowsPathSettings::~WindowsPathSettings()
{
}

/**
 * Creates a path separator.
 */
std::wstring WindowsPathSettings::GetPathSeparator() const
{
    return WindowsPath::DirectorySeperator();
}

/**
 * Creates a path.
 */
IPathSharedPtr WindowsPathSettings::CreatePath(const std::wstring& path) const
{
    return WindowsPath::Make(path);
}

/**
 * Creates a path builder.
 */
IPathBuilderSharedPtr WindowsPathSettings::CreatePathBuilder() const
{
    return PathBuilder::Make();
}
